pkg/cli/cmd/taskrun/list.go

```diff
@@ -0,0 +1,267 @@
+{{ end -}}{{- end -}}{{- end -}}
+{{- end -}}`
+
+// listCommand initializes a cobra command to list TaskRuns
+func listCommand(p common.Params) *cobra.Command {
+	opts := &options.ListOptions{
+		Limit:         50, // Default to 50
+		AllNamespaces: false,
+		SinglePage:    true, // Default to true
+	}
+
+	eg := `List all TaskRuns in a namespace 'foo':
+    tkn-results taskrun list -n foo
+
+List all TaskRuns in 'default' namespace:
+    tkn-results taskrun list -n default
+
+List TaskRuns with a specific label:
+    tkn-results taskrun list -L app=myapp
+
+List TaskRuns with multiple labels:
+    tkn-results taskrun list --label app=myapp,env=prod
+
+List TaskRuns from all namespaces:
+    tkn-results taskrun list -A
+
+List TaskRuns with limit of 20 per page:
+    tkn-results taskrun list --limit=20
+
+List TaskRuns for a specific task:
+    tkn-results taskrun list foo -n namespace
+
+List TaskRuns with partial task name match:
+    tkn-results taskrun list build -n namespace
+
+List TaskRuns for a specific PipelineRun:
+    tkn-results taskrun list --pipelinerun my-pipeline-run -n namespace
+`
+	cmd := &cobra.Command{
+		Use:     "list [task-name]",
+		Aliases: []string{"ls"},
+		Short:   "List TaskRuns in a namespace",
+		Annotations: map[string]string{
+			"commandType": "main",
+		},
+		Example: eg,
+		PreRunE: func(_ *cobra.Command, args []string) error {
+			c, err := config.NewConfig(p)
+			if err != nil {
+				return err
+			}
+			opts.Client, err = client.NewRESTClient(c.Get())
+			if err != nil {
+				return err
+			}
+
+			if opts.Limit < 5 || opts.Limit > 1000 {
+				return errors.New("limit should be between 5 and 1000")
+			}
+
+			// Validate label format if provided
+			if opts.Label != "" {
+				return common.ValidateLabels(opts.Label)
+			}
+
+			if len(args) > 0 {
+				opts.ResourceName = args[0]
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return listTaskRuns(cmd.Context(), p, opts)
+		},
+	}
+
+	cmd.Flags().Int32VarP(&opts.Limit, "limit", "", 50, "Maximum number of TaskRuns to return (must be between 5 and 1000, default is 50)")
+	cmd.Flags().BoolVarP(&opts.AllNamespaces, "all-namespaces", "A", false, "List TaskRuns from all namespaces")
+	cmd.Flags().StringVarP(&opts.Label, "label", "L", "", "Filter by label (format: key=value[,key=value...])")
+	cmd.Flags().StringVarP(&opts.PipelineRun, "pipelinerun", "", "", "Filter TaskRuns by PipelineRun name. Note that multiple PipelineRuns can have the same name, so this will return TaskRuns from all PipelineRuns with the matching name.")
+	cmd.Flags().BoolVar(&opts.SinglePage, "single-page", true, "Return only a single page of results")
+
+	return cmd
+}
+
+func listTaskRuns(ctx context.Context, p common.Params, opts *options.ListOptions) error {
+	// Build filter string
+	filter := common.BuildFilterString(opts, "taskrun")
+
+	// Handle all namespaces
+	parent := fmt.Sprintf("%s/results/-", p.Namespace())
+	if opts.AllNamespaces {
```

Comment:
So we give precedence to -A if specified with -n. Do we have a check to verify these are not specified together?

Reply:
I can add the check to not to use both the flags at the same time. 
But`p.Namespace()` will have the namespace which user is already using even if he is not specifying the `-n` flag

So I am just thinking if adding any check really makes sense.

Reply:
I have added a error message when user uses both the flags in the same command